rsts/deployment/configuration/general.rst

```diff
@@ -304,10 +305,60 @@ of Flyte's task execution. It ensures complete control over Pod configuration,
 supporting all options available through the resource and ensuring maintainability
 in future versions.
 
-To initialize a default PodTemplate in Flyte:
+Starting with the Flyte 1.4 release, we now have 2 ways of defining `PodTemplate <https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates>`__:
+1. Compile-time PodTemplate defined at the task level
+2. Runtime PodTemplates
+
+
+Compile-time PodTemplate usage
+==============================
+
+We can define a compile-time pod template, as part of the definition of a `Task <https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.task.html#flytekit-task>`__, for example:
+
+.. code-block:: python
+
+    @task(
+        pod_template=PodTemplate(
+            primary_container_name="primary",
+            labels={"lKeyA": "lValA", "lKeyB": "lValB"},
+            annotations={"aKeyA": "aValA", "aKeyB": "aValB"},
+            pod_spec=V1PodSpec(
+                containers=[
+                    V1Container(
+                        name="primary",
+                        image="repo/placeholderImage:0.0.0",
+                        command="echo",
+                        args=["wow"],
+                        resources=V1ResourceRequirements(limits={"cpu": "999", "gpu": "999"}),
+                        env=[V1EnvVar(name="eKeyC", value="eValC"), V1EnvVar(name="eKeyD", value="eValD")],
+                    ),
+                ],
+                volumes=[V1Volume(name="volume")],
+                tolerations=[
+                    V1Toleration(
+                        key="num-gpus",
+                        operator="Equal",
+                        value=1,
+                        effect="NoSchedule",
+                    ),
+                ],
+            )
+        )
+    )
+    def t1() -> int:
+        ...
+
+Notice how in this example we are defining a new PodTemplate inline, which allows us to define a full
+`V1PodSpec <https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1PodSpec.md>`__ and also define
+the name of the primary container, labels, and annotations.
+
+The term compile-time here refers to the fact that the pod template definition is part of the `TaskSpec <https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#ref-flyteidl-admin-taskclosure>`__.
+
+Runtime PodTemplate
+====================
 
 Set the ``default-pod-template-name`` in FlytePropeller
-========================================================
+--------------------------------------------------------
```

Comment:
The `Set the default-pod-template-name in FlytePropeller` above this line is only required when using a default PodTemplate and not if it is set in the task decorator. Does it make sense to make this distinction?

Reply:
I'm going to add a little blurb explaining the difference between the default Runtime PodTemplate and a named one.